sky/global_user_state.py

```diff
@@ -159,7 +162,15 @@ def add_or_update_cluster(cluster_name: str,
             cluster_name,
             ClusterStatus.STOPPED.value,
         ))
+
+    if not metadata:
+        metadata = {}
+        metadata['usage_intervals'] = []
+        metadata['cost_before_start'] = 0
+        metadata['latest_queried_cost'] = 0
```

Comment:
Is there a particular reason to store this state in the db instead of computing the cost from `usage_intervals` when a query is executed?

I'm leaning towards not storing any running cost related metadata in the db because it is continuously changing and has chances of introducing bugs if this state is not updated before querying. Instead, we should probably have a `get_cost(handle)` method that looks at the `usage_intervals` and computes the cost when requested. wdyt?